formula-coroutines/src/main/java/com/instacart/formula/coroutines/FlowAction.kt

```diff
@@ -51,31 +50,32 @@ interface FlowAction<Event> : Action<Event> {
          *
          * @param key Used to distinguish this [Action] from other actions.
          */
-        inline fun <Event> fromFlow(
-            scope: CoroutineScope = MainScope(),
+        fun <Event> fromFlow(
             key: Any?,
-            crossinline create: () -> Flow<Event>
+            create: () -> Flow<Event>
         ): Action<Event> {
-            return object : FlowAction<Event> {
-                override val scope: CoroutineScope = scope
-
-                override fun flow(): Flow<Event> {
-                    return create()
-                }
-
-                override fun key(): Any? = key
-            }
+            return FlowActionImpl(key, create)
         }
     }
 
     fun flow(): Flow<Event>
 
-    val scope: CoroutineScope
-
+    @OptIn(DelicateCoroutinesApi::class)
```

Comment:
Interesting Annotation. Looks like it's just the `GlobalScope` usage that requires this?

Reply:
That's correct